Sources/Workspace/Workspace.swift

```diff
@@ -624,10 +624,15 @@ public class Workspace {
         // If we already have it, fetch to update the repo from its remote.
         if let dependency = managedDependencies[repository] {
             let path = checkoutsPath.appending(dependency.subpath)
-            // Fetch the checkout in case there are updates available.
-            let workingRepo = try repositoryManager.provider.openCheckout(at: path)
-            try workingRepo.fetch()
-            return path
+            
+            // Make sure the directory is not missing (we will have to clone again
+            // if not).
+            if fileSystem.isDirectory(path) {
```

Comment:
I mean, convert this if statement into an assert because we always expect this directory to be present.

```swift
assert(fileSystem.isDirectory(path))
// Fetch the checkout in case there are updates available.
let workingRepo = try repositoryManager.provider.openCheckout(at: path)
try workingRepo.fetch()
return path
```